Source/Core/VideoCommon/GraphicsModSystem/Runtime/CustomAssetLoader.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "VideoCommon/GraphicsModSystem/Runtime/CustomAssetLoader.h"
+
+#include "Common/Logging/Log.h"
+#include "Common/MemoryUtil.h"
+#include "Common/StringUtil.h"
+#include "VideoCommon/GraphicsModSystem/Runtime/CustomAssetLibrary.h"
+
+namespace VideoCommon
+{
+void CustomAssetLoader::Init()
+{
+  m_file_monitor_thread_shutdown.Clear();
+
+  const size_t sys_mem = Common::MemPhysical();
+  const size_t recommended_min_mem = 2 * size_t(1024 * 1024 * 1024);
+  // keep 2GB memory for system stability if system RAM is 4GB+ - use half of memory in other cases
+  m_max_memory_available =
+      (sys_mem / 2 < recommended_min_mem) ? (sys_mem / 2) : (sys_mem - recommended_min_mem);
```

Comment:
This is copied from the existing custom texture logic, right? I assume the two will never be active at the same time?

Reply:
Yes it is copied.  The _hope_ is that the custom texture logic will end up using this functionality (both so that it can gain the asynchronous features this provides and so that we have one place for all texture operations).  If not, I'll probably have to expose some different way to set max memory size.